include/rmm/mr/device/system_memory_resource.hpp

```diff
@@ -114,8 +114,14 @@ class system_memory_resource final : public device_memory_resource {
    */
   void do_deallocate(void* ptr,
                      [[maybe_unused]] std::size_t bytes,
-                     [[maybe_unused]] cuda_stream_view stream) override
+                     cuda_stream_view stream) override
   {
+    // With `cudaFree`, the CUDA runtime keeps track of dependent operations and does implicit
```

Comment:
Please mention in the docstring of this function that it synchronizes `stream`.

Reply:
Done.